app/components/todos-route/component.js

```diff
@@ -1,38 +1,35 @@
-// controllers/todos.js
 import Ember from 'ember';
 
-var isEmpty  = Ember.isEmpty;
+var isEmpty = Ember.isEmpty;
 var filterBy = Ember.computed.filterBy;
-var notEmpty = Ember.computed.notEmpty;
-
-export default Ember.ArrayController.extend({
-  active:    filterBy('@this', 'isCompleted', false),
-  completed: filterBy('@this', 'isCompleted', true),
-  state: 'all',
-  hasCompleted: notEmpty('completed.[]'),
-  queryParams: [
-    'state'
-  ],
-
-  inflection: function () {
+var computed = Ember.computed;
+
+export default Ember.Component.extend({
+  completed: filterBy('todos', 'isCompleted', true),
+  active: filterBy('todos', 'isCompleted', false),
+
+  inflection: computed('active.[]', function() {
     var active = this.get('active.length');
     return active === 1 ? 'item' : 'items';
-  }.property('active.[]'),
+  }).readOnly(),
 
-  allAreDone: function (key, value) {
+  allAreDone: computed('@each.isCompleted', function (key, value) {
     if (arguments.length === 2) {
-      this.setEach('isCompleted', value);
-      this.invoke('save');
+      // TODO: use action instead of a 2 way CP.
+      var todos = this.get('active');
```

Comment:
Only selecting active todos breaks clearing completed state for all todos. You probably want something like this:
```javascript
var todos = this.get(value ? 'active' : 'completed');
```